core/server/services/auth/session/middleware.js

```diff
@@ -20,11 +20,11 @@ const getOrigin = (req) => {
         return origin;
     }
 
-    try {
-        return new URL(req.get('referrer')).origin;
-    } catch (e) {
-        return null;
+    const {protocol, host} = parseUrl(referrer);
```

Comment:
What i wonder in general:
As far as i know, browsers do not send the `Origin` header for GET requests and you can disable HTTP referrer in your browser, because you don't want to expose it.

Reply:
The `Origin` is not always present - it varies from browser, which is why we fall back to the `Referer` if it's not there. We've followed the guidelines in the [OWASP CSRF prevention document](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet#What_to_do_when_Both_Origin_and_Referer_Headers_Aren.27t_Present) and they suggest blocking requests that contain neither header.

That is a good point about disabling the Referer header but we should probably make that a separate issue - as this is just a bug fix of existing specced behaviour

Reply:
Yeah def separate issue 👍 Was just wondering about this in general :)

Reply:
Yeah :) I hadn't thought about it at all - i'll talk with kevin about it next week!